discovery/azure/azure.go

```diff
@@ -493,13 +493,12 @@ func (client *azureClient) getNetworkInterfaceByID(networkInterfaceID string) (n
 
 	resp, err := client.nic.GetSender(req)
 	if err != nil {
-		result.Response = autorest.Response{Response: resp}
 		return result, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure sending request")
 	}
 
 	result, err = client.nic.GetResponder(resp)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
+		return result, autorest.NewErrorWithError(err, "network.InterfacesClient", "Get", resp, "Failure responding to request")
```

Comment:
It's usual to return nil for the result if there's an error.

Reply:
I just wanted to be consistent with the style of the rest of the file.

What do you mean? Should I fix it in just this place or in the entire file?

Reply:
Looking at the `github.com/prometheus/prometheus/discovery/azure` package, there are quite a few places where we could do `return nil, err`. I'd lean toward "fixing" these parts in a separate PR.

Reply:
So whats the status then?

Should this be merged or are we waiting on the follow-up change?